### PR TITLE
Initial Repo Setup

### PR DESCRIPTION
**Pull request by garethellis0**
- added a sample_package
- set up Continuous Integration

- **Note:** that the merged CI tests are failing because the master branch is missing files required for dependency installation. We'll have to let this one in with failing tests, but this **should** be the only time.

### CODE REVIEW

**Comment by Dulluhan** [comment]
For the problem with the master branch failing, out of curiosity, how does CI find the dependencies that are not listed. Since according to docs the rest of our ros-dependencies should be done via rosdep.

File: install_dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+#######################################################################
+# STOP: If the dependency you want to add is required for the project #
```
Comment (Dulluhan) [inline install_dependencies.sh]: This should probably go in the README as well

**Comment by garethellis0** [comment]
Yeah, the issue with master failing basically comes down to the `.rosinstall` file, which indicates the root of a workspace for rosdep, not being on the master branch. I'm honestly not too sure why it needs to be on both branches (the CI scripts merge it in from master to the current branch), but we're using a CI system forked from a major ROS package (moveit), so I thought it best to leave it in.

File: install_dependencies.sh (hunk 1 above)
Comment (garethellis0) [reply install_dependencies.sh]: Hmm, it's a pretty standard ROS feature. While I agree that people should know that,  there are lots of other things in ROS that are just as important, and I don't think we want to put them in, so I'm inclined to leave this out of the README for now.

**Pull request merged**